### PR TITLE
Upkeep for parse tools

### PR DESCRIPTION
**Pull request by lionel-**
- Remove `r_` prefix, reexport from top-level, and use `harp::` prefix at call sites
- Reorganise consistently in `parse.rs`, `eval.rs`, and `source.rs`
- Rename `parse_vector()` to `parse_status()`.
- Add `parse_eval_global()` and `parse_eval_base()` for tests

### CODE REVIEW

File: crates/ark/src/lsp/completions/sources/unique/custom.rs [hunk 1]
```diff
@@ -9,8 +9,8 @@ use anyhow::Result;
 use harp::exec::RFunction;
 use harp::exec::RFunctionExt;
 use harp::object::RObject;
-use harp::utils::sym_quote_invalid;
 use harp::utils::r_typeof;
+use harp::utils::sym_quote_invalid;
```
Comment (DavisVaughan) [inline crates/ark/src/lsp/completions/sources/unique/custom.rs:13]: Why did it move? 🤔

File: crates/harp/src/parse.rs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+use crate::try_catch;
+use crate::RObject;
+
+pub enum ParseResult {
+    Complete(SEXP),
+    Incomplete,
+}
+
+/// Returns a single expression
+pub fn parse(code: &str) -> crate::Result<RObject> {
```
Comment (DavisVaughan) [inline crates/harp/src/parse.rs:27]:
```suggestion
pub fn parse_expr(code: &str) -> crate::Result<RObject> {
```

?

File: crates/harp/src/parse.rs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        let code = r_string!(code, protect);
+
+        RFunction::new("base", "parse")
+            .param("text", code)
+            .param("keep.source", true)
+            .call()
+    }
+}
+
+#[allow(non_upper_case_globals)]
```
Comment (DavisVaughan) [inline crates/harp/src/parse.rs:75]: still needed?

File: crates/harp/src/parse.rs [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+pub unsafe fn parse_status(code: &str) -> crate::Result<ParseResult> {
```
Comment (DavisVaughan) [inline crates/harp/src/parse.rs:76]: Propagating unsafety here but not in the other helpers?

**Review by DavisVaughan — APPROVED** [review]

File: crates/ark/src/lsp/completions/sources/unique/custom.rs (hunk 1 above)
Comment (lionel-) [reply crates/ark/src/lsp/completions/sources/unique/custom.rs:13]: No idea! The formatter wants it there though.

**Pull request merged**